src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -118,6 +119,13 @@ public class LibraryTab extends Tab {
     private enum PanelMode { MAIN_TABLE, MAIN_TABLE_AND_ENTRY_EDITOR }
 
     private static final Logger LOGGER = LoggerFactory.getLogger(LibraryTab.class);
+
+    /**
+     * The interval in seconds between Git status checks.
+     * 10 seconds provides a good balance between UI responsiveness and reducing system load.
+     */
+    private static final int GIT_STATUS_REFRESH_INTERVAL_SECONDS = 10;
```

Comment:
A better approach could be to add af file watcher to `.git/index` to move from pull-based to push-based code.